File: app/src/main/java/com/example/androidtvapptutorial/MainFragment.java
package com.example.androidtvapptutorial;

import android.os.Bundle;
import android.util.Log;
import androidx.leanback.app.BrowseSupportFragment;
import androidx.leanback.widget.ArrayObjectAdapter;
import androidx.leanback.widget.HeaderItem;
import androidx.leanback.widget.ListRowPresenter;
import androidx.leanback.widget.PageRow;

import java.util.ArrayList;
import java.util.List;

//import androidx.leanback.app.BrowseFragment;

public class MainFragment extends BrowseSupportFragment {

    private final String TAG = MainFragment.class.getSimpleName();

    private static SimpleBackgroundManager simpleBackgroundManager = null;

    private ArrayObjectAdapter mRowsAdapter;
    private static final int GRID_ITEM_WIDTH = 300;
    private static final int GRID_ITEM_HEIGHT = 200;

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        Log.i(TAG, "onActivityCreated");
        super.onActivityCreated(savedInstanceState);

        prepareBackgroundManager();

        setupUIElements();

        loadData();

        getMainFragmentRegistry().registerFragment(PageRow.class,new RowFragmentFactory());
    }

    private void prepareBackgroundManager(){
        simpleBackgroundManager = new SimpleBackgroundManager(getActivity());
    }

    private void setupUIElements() {
        // setBadgeDrawable(getActivity().getResources().getDrawable(R.drawable.videos_by_google_banner));
        setTitle("Hello Android TV!"); // Badge, when set, takes precedent
        /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            setBadgeDrawable(getResources().getDrawable(R.mipmap.videos_by_google_banner,getContext().getTheme()));
        }*/
        // over title
        setHeadersState(HEADERS_ENABLED);
        setHeadersTransitionOnBackEnabled(true);

        // set fastLane (or headers) background color
        setBrandColor(getResources().getColor(R.color.fastlane_background));
        // set search icon color
        setSearchAffordanceColor(getResources().getColor(R.color.search_opaque));
        /*
        setHeaderPresenterSelector(new PresenterSelector() {
            @Override
            public Presenter getPresenter(Object o) {
                return new IconHeaderItemPresenter();
            }
        });
        */

        prepareEntranceTransition();
    }

    private void loadData(){
        //Create Array Object Adapter Instance and Set Adapter to Browse Rows
        mRowsAdapter = new ArrayObjectAdapter(new ListRowPresenter());
        setAdapter(mRowsAdapter);

        //Create List of HeaderItem Data and add to the Instance of Array Object Adapter
        List<HeaderItem> dummyHeaders = new ArrayList<>();
        dummyHeaders.add(new HeaderItem(0, "USB All"));
        dummyHeaders.add(new HeaderItem(0, "USB1"));
        dummyHeaders.add(new HeaderItem(1, "USB2"));
        dummyHeaders.add(new HeaderItem(2, "USB3"));
        for (HeaderItem header : dummyHeaders) {
            mRowsAdapter.add(new PageRow(header));
        }

        //After the Fragment finishes the loading data it will execute an entrance transition
        startEntranceTransition();
    }

}

File: app/src/main/java/com/example/androidtvapptutorial/kotlin/MainFragment.kt
/*
package com.example.androidtvapptutorial

import android.content.res.Resources
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.support.v17.leanback.app.BrowseFragment
import android.support.v17.leanback.widget.*
import android.util.Log
import android.view.Gravity
import android.view.ViewGroup
import android.widget.TextView
import android.support.v17.leanback.widget.ListRowPresenter
import android.support.v17.leanback.widget.ArrayObjectAdapter
import android.support.v17.leanback.widget.ListRow







class MainFragment : BrowseFragment() {
    private val TAG = MainFragment::class.java.simpleName

    private lateinit var mRowsAdapter: ArrayObjectAdapter
    private val GRID_ITEM_WIDTH = 300
    private val GRID_ITEM_HEIGHT = 200

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        Log.i(TAG, "onActivityCreated")
        super.onActivityCreated(savedInstanceState)

        setupElements()

        loadRows()
    }

    private fun setupElements(){
        setTitle("Hello Android TV!")
        */
/*
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            setBadgeDrawable(resources.getDrawable(R.mipmap.ic_launcher,context.theme))
        }
        *//*

        setHeadersState(HEADERS_ENABLED);
        setHeadersTransitionOnBackEnabled(true);
        setBrandColor(getResources().getColor(R.color.fastlane_background));
        setSearchAffordanceColor(getResources().getColor(R.color.search_opaque));
    }

    private fun loadRows() {
        mRowsAdapter = ArrayObjectAdapter(ListRowPresenter())

        val gridItemPresenterHeader : HeaderItem = HeaderItem(0,"GridItemPresenter")
        val mGridItemPresenter : GridItemPresenter = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            GridItemPresenter(GRID_ITEM_WIDTH,GRID_ITEM_HEIGHT,getResources().getColor(R.color.default_background,context.getTheme()))
        } else {
            GridItemPresenter(GRID_ITEM_WIDTH,GRID_ITEM_HEIGHT,getResources().getColor(R.color.default_background))
        }
        val gridRowAdapter : ArrayObjectAdapter = ArrayObjectAdapter(mGridItemPresenter)
        gridRowAdapter.add("Item 1")
        gridRowAdapter.add("Item 2")
        gridRowAdapter.add("Item 3")
        mRowsAdapter.add(ListRow(gridItemPresenterHeader, gridRowAdapter))

        // set
        setAdapter(mRowsAdapter)
    }
}*/
